be/src/olap/rowset/beta_rowset.cpp

```diff
@@ -69,11 +69,22 @@ Status BetaRowset::init() {
 }
 
 Status BetaRowset::do_load(bool /*use_cache*/) {
-    // do nothing.
     // the segments in this rowset will be loaded by calling load_segments() explicitly.
+    auto segment_count = num_segments();
```

Comment:
不要在这改，这里改有问题。
    // do not use cache to load index
    // because the index file may conflict
    // and the cached fd may be invalid
    RETURN_IF_ERROR(org_rowset->load(false));
有一些备份恢复的地方会调用这个函数，此时就会带来很多不必要的IO。
我们给get_segment_num_rows 这个函数加一个lock，让他去加载，然后返回。
然后这个do load 函数，可能没啥用了，你顺道删了把。之前的实现都是空的。